tsdb/engine/tsm1/compact.go

```diff
@@ -762,6 +759,36 @@ func (c *Compactor) write(path string, iter KeyIterator) (err error) {
 	return nil
 }
 
+func (c *Compactor) add(files []string) bool {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	var inuse bool
+	for _, f := range files {
+		if _, ok := c.files[f]; ok {
+			inuse = true
```

Comment:
Yeah.  I'll rework.  I had a different approach initially and this is leftover from that.